### PR TITLE
Improving Assembly and adding a 3rd ZIP file with only the core

### LINKED ISSUE #388: The assembly should include the needed code instead of excluding what's not needed
Today we have the `assembly.xml` file that includes everything, and then excludes parts of the workshop:

```xml
<exclude>rest-fights/**</exclude>
<exclude>rest-heroes/**</exclude>
<exclude>rest-villains/**</exclude>
```

We should do the other way round. We should include only once needed

### PR DESCRIPTION
**Pull request by agoncal**
I only include files instead of excluding files

Fix #388

### CODE REVIEW

File: quarkus-workshop-super-heroes/assembly-core.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                <include>super-heroes/rest-heroes/.mvn/wrapper/maven-wrapper.properties</include>
+                <include>super-heroes/rest-heroes/src/**/*</include>
+
+                <include>super-heroes/rest-villains/*</include>
+                <include>super-heroes/rest-villains/.mvn/wrapper/maven-wrapper.properties</include>
+                <include>super-heroes/rest-villains/src/**/*</include>
+
+                <include>super-heroes/ui-super-heroes/.mvn/wrapper/maven-wrapper.properties</include>
+                <include>super-heroes/ui-super-heroes/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/docker/**/*</include>
```
Comment (holly-cummins) [inline quarkus-workshop-super-heroes/assembly-core.xml:56]: I think it would be safe to include everything in `/src`, wouldn't it?

File: quarkus-workshop-super-heroes/assembly.xml [hunk 2]
```diff
@@ -24,47 +24,29 @@
...
+                <include>super-heroes/kubernetes/**/*</include>
+
+                <include>super-heroes/ui-super-heroes/.mvn/wrapper/maven-wrapper.properties</include>
+                <include>super-heroes/ui-super-heroes/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/docker/**/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/java/**/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/resources/**/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/webui/*</include>
+                <include>super-heroes/ui-super-heroes/src/main/webui/src/**/*</include>
+                <include>super-heroes/ui-super-heroes/src/test/**/*</include>
```
Comment (holly-cummins) [inline quarkus-workshop-super-heroes/assembly.xml:49]: As above, I think we could shorten this to  <include>super-heroes/ui-super-heroes/src/**

Comment (holly-cummins) [reply quarkus-workshop-super-heroes/assembly.xml:49]: Oh wait, I see, that might get the `node_modules` folder and other quinoa stuff, but we could add an explicit exclude for those.

File: quarkus-workshop-super-heroes/assembly-core.xml (hunk 1 above)
Comment (agoncal) [reply quarkus-workshop-super-heroes/assembly-core.xml:56]: Well, I thought it would, but I ended up with a `node_module` directory just under `src`. I don't know if it's expected or not (there is already a `node_module` directory under `.quinoa`) but to be safe, I did it this way avoiding to include a huge `node_module` directory

File: quarkus-workshop-super-heroes/assembly.xml (hunk 2 above)
Comment (agoncal) [reply quarkus-workshop-super-heroes/assembly.xml:49]: Yes, I found out that there was a `node_modules` directory under `src`. Which I found strange. 

I tried having an `exclude` but I couldn't make it work (I thought `excludes` took precedence over `includes` but couldn't make it work)

File: quarkus-workshop-super-heroes/assembly-core.xml (hunk 1 above)
Comment (holly-cummins) [reply quarkus-workshop-super-heroes/assembly-core.xml:56]: Well, with an `<exclude>` for **/node_modules/** as we had before. We can probably keep quite a few of the earlier excludes, actually, things like `<exclude>**/ObjectStore</exclude>`. Then it's kind of a belt-and-braces approach where we include generously, and filter out stuff we shouldn't have.

**Comment by agoncal** [comment]
@holly-cummins yes, the workshop starter file shrank because I removed the `load-super-heroes`. One day we will turn it into a Quarkus CLI so we can add a new chapter "how to create a CLI with Quarkus"

File: quarkus-workshop-super-heroes/assembly-core.xml (hunk 1 above)
Comment (agoncal) [reply quarkus-workshop-super-heroes/assembly-core.xml:56]: Ok, let me have another look at the exclude

Comment (holly-cummins) [reply quarkus-workshop-super-heroes/assembly-core.xml]: Oh, very nice! That should be much more robust than what we have now.
